pkg/reconciler/pipelinerun/resources/resultrefresolution.go

```diff
@@ -124,6 +124,18 @@ func removeDup(refs ResolvedResultRefs) ResolvedResultRefs {
 // convertParamsToResultRefs converts all params of the resolved pipeline run task
 func convertParamsToResultRefs(pipelineRunState PipelineRunState, target *ResolvedPipelineRunTask) (ResolvedResultRefs, error) {
 	var resolvedParams ResolvedResultRefs
+	for _, condition := range target.PipelineTask.Conditions {
+		for _, param := range condition.Params {
+			resolvedResultRefs, err := extractResultRefsForParam(pipelineRunState, param)
+			if err != nil {
+				return nil, fmt.Errorf("unable to find result referenced by param %q in pipeline task %q: %w", param.Name, target.PipelineTask.Name, err)
```

Comment:
nit: maybe indicate in the message here that it's `condition param %q`?

Reply:
good point. Fixing it.

Reply:
Done. Tests are running.

Reply:
If I reuse code, I will have to simplify the error message.

Reply:
refactored